src/history.cpp

```diff
@@ -1514,9 +1520,16 @@ bool history_t::search(history_search_type_t search_type, const wcstring_list_t
     }
 
     // Output any items we collected (which only happens in reverse).
-    for (auto iter = collected.rbegin(); iter != collected.rend(); ++iter) {
-        streams.out.append(*iter);
+    for (auto iter = collected.rbegin(); !output_error && iter != collected.rend(); ++iter) {
+        if (!streams.out.append(*iter)) {
+            // Don't force an error if output was aborted (typically via Ctrl-C/SIGINT); just don't
+            // try writing any more.
+            output_error = true;
+        }
     }
+
+    // We are intentionally not returning false in case of an output error, as the user aborting the
+    // output early (the most common case) isn't a reason to exit w/ a non-zero status code.
```

Comment:
.. but in this case we likely exit with a non-zero signal code like SIGINT = 130